similarity_processor/similarity_io.py

```diff
@@ -120,56 +132,53 @@ def __write_xlsx(self, data_f, name):
         data_f.to_excel(writer, sheet_name=name)
         writer.save()
 
+    def __write_csv(self, data_f, name):
+        """ Function which write the dataframe to xlsx """
+        file_path = os.path.join(self.__get_file_path(), self.__get_file_name() + "_" + name)
+        data_f.to_csv(file_path, header=True)
+
     def __new_text_df(self):
         """ Function which is created to form the new dataframe to include new text if
         entered in UI """
         __new_df = pd.DataFrame({self.uniq_header: ["New/ID_TBD"], 'Steps': [self.new_text]})
         self.data_frame = __new_df.append(self.data_frame, ignore_index=True)
 
     def __process_cos_match(self):
-        """ Function which process the dataframe for matching/finding similarity index """
-        # list to store the analysis
-        __reco = []
-        # Loop through the set of rows in the dataframe
-        for master_text_index in self.data_frame.index:
-            __uniq_id = self.data_frame[self.uniq_header][master_text_index]
-            master_text = self.data_frame['Steps'][master_text_index]
-            vector1 = text_to_vector(str(master_text))
-            # Loop through the rows immediate next from the initially selected in previous loop
-            for match_text_index in range(master_text_index + 1, len(self.data_frame.index)):
-                match_id = self.data_frame[self.uniq_header][match_text_index]
-                match_text = self.data_frame['Steps'][match_text_index]
-                vector2 = text_to_vector(str(match_text))
-                # Generate the cosine similarity match value
-                cosine = get_cosine(vector1, vector2)
-                __reco.append({self.uniq_header: str(__uniq_id), 'Potential Match':
-                               str(match_id), 'Similarity Index': cosine * 100})
-            if self.is_new_text == 1:
-                break
-        # Create dataframe and write
-        self.__write_xlsx(pd.DataFrame(__reco), "recomendation")
+        """ Function which process the data frame for matching/finding similarity index """
+        count_vect = CountVectorizer()
+        word_count_vector = count_vect.fit_transform(self.data_frame['Steps'].to_numpy())
+        c_sim = 100 * (cosine_similarity(word_count_vector))
+        self.data_frame["Potential Match"] = self.data_frame[self.uniq_header]
```

Comment:
Nitpick - please see if this is valuable. 

Sometimes the key is referred with double quotes, and sometimes with single-quotes. Should we standardize it?

Reply:
standardized it with " ", but in some cases this in not applicable if there are quote with in another.